operator/src/main/java/oracle/kubernetes/weblogic/domain/model/DomainResource.java

```diff
@@ -1196,24 +1208,25 @@ private void checkPortNameLength(V1ContainerPort port, String name, String prefi
 
     private void verifyClusterResourcesNotInUse(
         KubernetesResourceLookup kubernetesResources) {
-      getSpec().getClusters().forEach(cluster ->
-          Optional.ofNullable(kubernetesResources.findCluster(cluster))
-              .ifPresent(cr -> verifyClusterInUseByAnotherDomain(kubernetesResources, cr)));
+      Optional.ofNullable(getSpec().getClusters()).orElse(new ArrayList<>())
+          .forEach(reference ->
+              Optional.ofNullable(kubernetesResources.findClusterInNamespace(reference, getNamespace()))
+              .ifPresent(cluster -> verifyClusterInUseByAnotherDomain(kubernetesResources, cluster)));
```

Comment:
This is not in this change, but should this method name be verifyCluster**Not**InUseByAnotherDomain?

Reply:
The proposed name is more consistent with the pattern. will change.